examples/simpack_xrd_dyn_AlGaAs.py

```diff
@@ -0,0 +1,178 @@
+
+
+# definition of heterostructure layers
+
+sub = xu.simpack.Layer(GaAs_, float('inf'))
+AlGaAs80_1 = xu.simpack.Layer(AlGaAs80, 20000, relaxation=0.)
+GaAs_2 = xu.simpack.Layer(GaAs_, 2000, relaxation=0.)
+AlGaAs80_3 = xu.simpack.Layer(AlGaAs80, 18000, relaxation=0.)
+GaAs_4 = xu.simpack.Layer(GaAs_, 3200, relaxation=0.0)
+
+
+# heterostructure content
+heterostructure_ = AlGaAs80_1 \
+                  +GaAs_2 \
+                  +AlGaAs80_3\
+                  +GaAs_4
+
+# turned into a pseudomorphic structure grown along the 001 direction
+heterostructure = xu.simpack.PseudomorphicStack001('Hetereostructure', sub, heterostructure_)
+
+# X-Ray energy
+wavelength = xu.wavelength('CuKa1')
+en = xu.energy('CuKa1')
+
+# Monochromator
+thMono = np.arcsin(wavelength/(2 * xu.materials.Ge.planeDistance(2, 2, 0)))
+
+# Dynamical diffraction model
+md = xu.simpack.DynamicalModel(heterostructure, I0 = 4e5,
+                               background = 1,
+                               energy = en,
+                               resolution_width = 0.0015,
+                               Cmono = np.cos(2 * thMono),
+                               polarization ='both'
+                              )
+
+fitmdyn = xu.simpack.FitModel(md)
+fitmdyn.lmodel.set_hkl((0, 0, 4))
+
+fitmdyn.verbose = True
+fitmdyn.plot = True
+
+
+# set fit parameters
+
+# AlGaAs80_1, 'AlAs_0_80_GaAs_0_20__c' parameter name found using print(params) (see below)
+fitmdyn.set_param_hint('AlAs_0_80_GaAs_0_20__c', vary = True, min = AlGaAs80.a * 0.998, max = AlGaAs80.a * 1.002)
+fitmdyn.set_param_hint('AlAs_0_80_GaAs_0_20__thickness', vary = True, min = 19800, max = 20200)
+
+
+# AlGaAs80_3 is grown using the same cells as AlGaAs_1 hence same composition and growth rate
+fitmdyn.set_param_hint('AlAs_0_80_GaAs_0_20__1_c', vary = True, expr = 'AlAs_0_80_GaAs_0_20__c') # same composition as AlGaAs80_1
+fitmdyn.set_param_hint('AlAs_0_80_GaAs_0_20__1_thickness', vary = True, expr = 'AlAs_0_80_GaAs_0_20__thickness * 18000 / 20000') # ratio 18000/20000 as thickness ratio between layer 1
+
+#GaAs_2
+fitmdyn.set_param_hint('GaAs_1_thickness', vary = True, min = 1990, max = 2010)
+
+#GaAs_4
+fitmdyn.set_param_hint('GaAs_2_thickness', vary = True, expr = 'GaAs_1_thickness*3200/2000') # thickness ratio between GaAs_4 and GaAs_2
+
+fitmdyn.set_param_hint('background', vary= True, min = 0, max = 10)
+fitmdyn.set_param_hint('I0', vary= True, min = 3e5, max = 5e5)
+fitmdyn.set_param_hint('resolution_width', vary = True, min = 0.0012, max = 0.0015)
+
+params = fitmdyn.make_params()
+
+
+# choose your fitting method, default 'leastsq'
+fit_kws = {'method':'differential_evolution'}
+
+# option to avoid error during fit
+lmfit_kws = {'nan_policy':'omit'}
+
+# launch fitting procedure
+fitr = fitmdyn.fit(Int, params, om_off, fit_kws = fit_kws, lmfit_kws = lmfit_kws)
+
+
+# get Al composition for perpendicular lattice parameter
+x_Al = x_Al(fitr.params['AlAs_0_80_GaAs_0_20__c'], GaAs_.a, AlGaAs80)
```

Comment:
This overwrites our definition of the function and is therefore bad style. I guess the best would be to avoid the function and use `AlGaAs.ContentBsym` if posssible.

Reply:
done